src/storage/src/hummock/store/version.rs

```diff
@@ -73,7 +74,9 @@ impl StagingVersion {
         impl Iterator<Item = &SstableInfo> + 'a,
     ) {
         let overlapped_imms = self.imm.iter().filter(move |imm| {
-            compaction_group_id == imm.compaction_group_id()
+            (compaction_group_id
```

Comment:
Now that we always pass compaction_group_id as StaticCompactionGroupId::NewCompactionGroup, the compaction_group_id in method parameters seems useless?

Reply:
> Now that we always pass compaction_group_id as StaticCompactionGroupId::NewCompactionGroup, the compaction_group_id in method parameters seems useless? That's to say prune_overlap can no longer filter by compaction group.

used in test